### ISSUE TITLE
best way to round trip `Result`

### ISSUE THREAD

**Issue by thedavidmeister** [3Hren/msgpack-rust#244]
i get issues where the serialized result is `0` (index) but then the deserialize wants `Ok` (name)

i did something like this but it seems hacky:

```rust
#[derive(Deserialize)]
enum FakeResult<T, E> {
    Ok(T),
    Err(E),
}

...

            impl<D: $crate::serde::de::DeserializeOwned> std::convert::TryFrom<$crate::SerializedBytes> for Result<$t, D> {
                type Error = $crate::SerializedBytesError;
                fn try_from(sb: $crate::SerializedBytes) -> Result<Result<$t, D>, $crate::SerializedBytesError> {
                    match $crate::rmp_serde::from_read_ref(&sb.0) {
                        Ok(FakeResult::Ok(v)) => Ok(Ok(v)),
                        Ok(FakeResult::Err(e)) => Ok(Err(e)),
                        Err(e) => Err($crate::SerializedBytesError::FromBytes(e.to_string())),
                    }
                }
            }
```

(the last bit i just pulled out of a macro but you get the idea i think)

**Comment by kornelski** [comment]
Do you have a testcase for mis-serialized `Result`?

**Comment by thedavidmeister** [comment]
@kornelski i couldn't get any examples of `Result` round tripping and i didn't see any examples in the round trip tests in this crate... i could probably put something together

**Comment by kornelski** [comment]
It's probably same as #245

**Comment by joshtriplett** [comment]
@thedavidmeister I believe https://github.com/serde-rs/serde/pull/1888 (released in serde 1.0.116) should fix this.
